core/src/main/java/net/kyori/moonshine/MoonshineBuilder.java

```diff
@@ -0,0 +1,239 @@
+//
+// moonshine - A localisation library for Java.
+// Copyright (C) 2021 Mariell Hoversholm
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published
+// by the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+package net.kyori.moonshine;
+
+import io.leangen.geantyref.GenericTypeReflector;
+import io.leangen.geantyref.TypeToken;
+import java.lang.reflect.Proxy;
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.NavigableSet;
+import java.util.TreeSet;
+import javax.annotation.concurrent.NotThreadSafe;
+import net.kyori.moonshine.exception.scan.UnscannableMethodException;
+import net.kyori.moonshine.message.IMessageRenderer;
+import net.kyori.moonshine.message.IMessageSender;
+import net.kyori.moonshine.message.IMessageSource;
+import net.kyori.moonshine.placeholder.IPlaceholderResolver;
+import net.kyori.moonshine.receiver.IReceiverLocatorResolver;
+import net.kyori.moonshine.strategy.IPlaceholderResolverStrategy;
+import net.kyori.moonshine.util.Weighted;
+import org.checkerframework.common.returnsreceiver.qual.This;
+import org.checkerframework.dataflow.qual.Deterministic;
+import org.checkerframework.dataflow.qual.Pure;
+import org.checkerframework.dataflow.qual.SideEffectFree;
+
+@NotThreadSafe // Technically wrong, but this is a builder.
+public final class MoonshineBuilder {
+  private MoonshineBuilder() {
+  }
```

Comment:
As per Effective Java, you should also throw an `AssertionError` in private constructors (I guess this is also a bit of a style thing, so idk).

Reply:
It's definitely a style thing, and I don't particularly care if people make instances of them. It's there as a hinder and a marker that it shouldn't be done, but I'm not going out of my way to stop people doing bad decisions. This is also why there is rarely a null-check on `@NonNull` parameters.